include/envoy/http/header_map.h

```diff
@@ -440,6 +440,15 @@ class HeaderMap {
    */
   virtual void addCopy(const LowerCaseString& key, const std::string& value) PURE;
 
+  /**
+   * Appends data to header. If header already has a value, the string ',' is added between the
+   * existing value and data.
+   * @param key specifies the name of the header to append.
+   * @param value specifies the value of the header to add. If the string is a reference string the
+   * reference data is not copied. This uses the underlying HeaderString::append method.
```

Comment:
I'm confused. Doesn't the data always need to be copied if we are appending and have to use a delimiter?

Should we have `appendCopy`, `appendReference` etc. to match the other APIs? `ToHeader` doesn't seem to match from my read. WDYT?

Reply:
OK I looked in to this more. I was reading off of the docstring for `HeaderString::append`, which is called in this method. But yeah, once the delimiter is added data's copied.

I agree, I changed it to `append`. Do you think I should match API to `appendCopy`? I'm not sure if there's a need for `appendReference`. There's one use case of non-inline headers in ext authz filters for `headers_to_append`, and only copies needed for inline headers (xff, via, forwarded client cert, grpc timeout, and envoy ip tags)

Reply:
I have a mild preference to make it `appendCopy` just for 100% clarity and to match the other methods, but I don't feel strongly about it. Whatever you think is best.